lib/src/platform/src/platform.dart

```diff
@@ -18,27 +18,29 @@ class Platform {
   static final StreamsChannel streamsChannel =
       StreamsChannel('io.ably.flutter.stream', codec);
 
-  /// Initializing ably on platform side by invoking `register` platform method.
-  /// Register will clear any stale instances on platform.
-  static Future? _initializer;
+  /// This field will reset to true when the state is reset. This allows us to
+  /// if an app has been restarted, or a hot restart (not hot reload) has
+  /// happened.
+  static bool _shouldResetAblyClients = true;
 
-  static Future _initialize() async {
-    if (_initializer == null) {
+  /// Clears instances on the Platform side
+  static Future<void> _resetOldAblyClients() async {
+    if (_shouldResetAblyClients) {
       AblyMethodCallHandler(methodChannel);
-      _initializer = methodChannel.invokeMethod(PlatformMethod.registerAbly);
       BackgroundIsolateAndroidPlatform();
+      await methodChannel.invokeMethod(PlatformMethod.resetAblyClients);
+      _shouldResetAblyClients = false;
```

Comment:
IMO it's always safer to use a callback or a state object that informs the callers than maintaining the state yourself.  this might be out of scope of this PR though

Reply:
`_shouldResetAblyClients` is actually the **state** field. When the app restarts or hot-restarts, this state field is set to `true` automatically because the class is re-instantiated:
```dart
  static bool _shouldResetAblyClients = true;
```

I think the mechanics here is simpler than you think, but I'm not sure

Reply:
And in actuality, if we weren't using static fields here, the constructor would just be called again. So all this logic drops out completely. This has been done in a future PR: https://github.com/ably/ably-flutter/pull/235

Reply:
Perhaps the addition of some log output when you test `_shouldResetAblyClients` would help ascertain whether / how often these code paths get executed? I don't see a sign of any tests that verify this functionality, so I'm assuming you have to observe this with manual testing. In which case, perhaps you could document your test procedure for that? This is all very opaque to me from just looking at this code - so very hard to review.

Reply:
I actually used the debugger to test this, so I didn't use any log lines (hot restart/reload are only relevant for debug mode). However, I've attached a video to confirm that it works. This logic (and field) is being replaced with just the constructor in https://github.com/ably/ably-flutter/pull/235.

https://user-images.githubusercontent.com/24711048/142890615-6413d0ff-c651-4fd0-b5d4-f261a4e605c4.mp4

I can't think of a test procedure I could write. Just restart the app or trigger hot restart, and this will be called as expected. I **could** explain the difference between starting the app, hot restart and reload like I did in the video though, so I've done that in 72472a8392d6d2427cc9a3071648a9635686cdb9. However, going into how to do it will depend what tool you use. If you use the terminal (flutter run), android studio or vscode. 

In the terminal (flutter run), you'll get:
![Screenshot 2021-11-22 at 15 55 58](https://user-images.githubusercontent.com/24711048/142893383-5b1d5498-cf83-4078-a6f7-2152365d9315.png)

In IDE, saving triggers hot restart.
Pressing something triggers hot reload. (Android studio, its the bug/debug icon)
Pressing something else triggers hot restart. (its noticeably slow) its the "run again" button (see screenshot of button)
![CleanShot 2021-11-22 at 15 54 56](https://user-images.githubusercontent.com/24711048/142893225-30d5ad8e-c04b-4fc7-b819-6822d0f92af3.png)

Documenting this stuff will get outdated easily, and it's also relatively available information.